core/server/worker/src/main/java/alluxio/worker/shortcircuit/SecureShortcircuitReadServer.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.shortcircuit;
+
+import alluxio.conf.PropertyKey;
+import alluxio.conf.ServerConfiguration;
+import alluxio.worker.WorkerProcess;
+import com.esotericsoftware.minlog.Log;
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.epoll.EpollDomainSocketChannel;
+import io.netty.channel.epoll.EpollServerDomainSocketChannel;
+import io.netty.channel.unix.DomainSocketAddress;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Secure Shortcircuit Read Server is used to obtain an FileDescriptor object
+ * of a specific file given the blockId.
+ */
+public class SecureShortcircuitReadServer {
```

Comment:
Creating a new netty-server should not be necessary. I'm also seeing duplication of code in inbound handler for this new netty-server.

Ideally, you should define a new read/write handler types and utilize those in new APIs defined in `BlockWorkerServiceHandler`.

Please let me know wdyt.

@jiacheliu3 fyi.

Reply:
not sure if sending a FD via gRPC is possible.  probably not.  I guess domain sockets are required to transfer a FD between process.

Reply:
@ggezer @beinan @jiacheliu3 @yuzhu thanks for reviewing code. could you help me to figure out how to sending a FD via gRPC or it is impossible?